app/src/main/java/org/xbmc/kore/service/library/SyncMusic.java

```diff
@@ -348,13 +301,7 @@ public void onSuccess(ApiList<AudioType.DetailsSong> result) {
 
                 int totalArtistsCount  = 0;
                 // Save partial results to DB
-                ContentValues songValuesBatch[] = new ContentValues[items.size()];
-                for (int i = 0; i < items.size(); i++) {
-                    AudioType.DetailsSong song = items.get(i);
-                    songValuesBatch[i] = SyncUtils.contentValuesFromSong(hostId, song);
-                    totalArtistsCount += song.artistid.size();
-                }
-                contentResolver.bulkInsert(MediaContract.Songs.CONTENT_URI, songValuesBatch);
+                insertSongsItems(items, contentResolver);
```

Comment:
I believe that if this is applied as is, there will be duplicate code.

After this line, there's currently code to insert the `songsArtists`, which will be kept, as this PR doesn't delete it. But that code is also included in the `insertSongsItems` implementation.